### PR TITLE
Adds a print_applied method to QNodes

### PR DESCRIPTION
**Pull request by josh146**
**Context:** Currently, it is hard to debug QNodes to ensure that they are executing the correct quantum circuit, for instance, if they include classical branching/logic within the QNode.

**Description of the Change:** Adds a `print_applied` method to QNodes, that prints the operation and observable queue as last constructed.

Example:

```python
import pennylane as qml
from pennylane.templates.embeddings import AngleEmbedding
from pennylane.templates.layers import StronglyEntanglingLayers
from pennylane.init import strong_ent_layer_uniform

dev = qml.device('default.qubit', wires=2)

@qml.qnode(dev, interface="torch")
def circuit(weights, x=None):
    AngleEmbedding(x, [0,1])
    StronglyEntanglingLayers(weights=weights, wires=[0,1])
    return qml.expval(qml.PauliZ(0))

init_weights = strong_ent_layer_uniform(n_wires=2)

circuit(init_weights, x=[1., 2.])
circuit.print_applied()
```

Output:

```
Operations
==========
RX(1.0, wires=[0])
RX(2.0, wires=[1])
Rot(5.2934758701132685, 2.6339939891643005, 6.268309774410444, wires=[0])
Rot(2.599559228882876, 2.0359675113467888, 0.7480448828174393, wires=[1])
CNOT(wires=[0, 1])
CNOT(wires=[1, 0])

Observables
===========
expval(PauliZ(wires=[0]))
```

**Benefits:** Easier to debug complicated QNodes.

**Possible Drawbacks:** Since the queue is not constructed until evaluation, using `print_applied()` will prior to first evaluation will emit a warning message and no information.

**Related GitHub Issues:** n/a (was mentioned on the discussion forum).

### CODE REVIEW

File: pennylane/qnode.py [hunk 1]
```diff
@@ -131,6 +131,36 @@ def _append_op(self, op):
...
+
+        return_map = {
+            qml.operation.Expectation: "expval",
+            qml.operation.Variance: "var",
+            qml.operation.Sample: "sample"
+        }
+
+        print("\nObservables")
+        print("===========")
+        for op in self.ev:
```
Comment (johannesjmeyer) [inline pennylane/qnode.py]: Just out of curiosity, are the names of `self.queue` and `self.ev` changed in the new QNode? I think they are very suboptimal.

File: tests/test_qnode.py [hunk 2]
```diff
@@ -111,6 +114,53 @@ def test_op_successors_extracts_all_successors(self, qnode):
...
+        node = qml.QNode(circuit, mock_device)
+
+        # test before construction
+        f = io.StringIO()
+
+        with contextlib.redirect_stdout(f):
+            node.print_applied()
+            out = f.getvalue().strip()
+
+        assert out == "QNode has not yet been executed."
```
Comment (johannesjmeyer) [inline tests/test_qnode.py]: One could think of having this in a separate test, but I think two assert statements in one test are fine.

File: pennylane/qnode.py (hunk 1 above)
Comment (josh146) [reply pennylane/qnode.py]: They should be changed, yes; feel free to double check in #354

File: tests/test_qnode.py (hunk 2 above)
Comment (josh146) [reply tests/test_qnode.py]: Trying to keep it contained for now, as it will need to be moved to the new QNode :laughing:

**Pull request merged**